Detectors/TRD/reconstruction/src/DataReaderTask.cxx

```diff
@@ -37,18 +41,22 @@ void DataReaderTask::init(InitContext& ic)
   };
 
   ic.services().get<CallbackService>().set(CallbackService::Id::Stop, finishFunction);
+  mDataDesc = "RAWDATA";
 }
 
-void DataReaderTask::sendData(ProcessingContext& pc)
+void DataReaderTask::sendData(ProcessingContext& pc, bool blankframe)
 {
   // mReader.getParsedObjects(mTracklets,mDigits,mTriggers);
-  mReader.getParsedObjects(mTracklets, mDigits, mTriggers);
-
-  LOG(info) << "Sending data onwards with " << mDigits.size() << " Digits and " << mTracklets.size() << " Tracklets and " << mTriggers.size() << " Triggers";
-  pc.outputs().snapshot(Output{o2::header::gDataOriginTRD, "DIGITS", 0, Lifetime::Timeframe}, mDigits);
-  pc.outputs().snapshot(Output{o2::header::gDataOriginTRD, "TRACKLETS", 0, Lifetime::Timeframe}, mTracklets);
-  pc.outputs().snapshot(Output{o2::header::gDataOriginTRD, "TRKTRGRD", 0, Lifetime::Timeframe}, mTriggers);
-  //    pc.outputs().snapshot(Output{o2::header::gDataOriginTRD,"STATS",0,Lifetime::Timerframe},mStats);
+  if (!blankframe) {
+    mReader.buildDPLOutputs(pc); //getParsedObjectsandClear(mTracklets, mDigits, mTriggers);
+  } else {
+    //ensure the objects we are sending back are indeed blank.
+    LOG(info) << "Sending data onwards with " << mDigits.size() << " Digits and " << mTracklets.size() << " Tracklets and " << mTriggers.size() << " Triggers and blankframe:" << blankframe;
+    pc.outputs().snapshot(Output{o2::header::gDataOriginTRD, "DIGITS", 0, Lifetime::Timeframe}, mDigits);
+    pc.outputs().snapshot(Output{o2::header::gDataOriginTRD, "TRACKLETS", 0, Lifetime::Timeframe}, mTracklets);
+    pc.outputs().snapshot(Output{o2::header::gDataOriginTRD, "TRKTRGRD", 0, Lifetime::Timeframe}, mTriggers);
```

Comment:
if I get it right, now the sending of the real data is done only via `mReader.buildDPLOutputs` while the mDigits, mTracklets and mTriggers are always empty and used on;y to push an empty output?
If so, then better to get rid of these data members and use here local vectors

Reply:
ah good point. changing now.